recipe/build-llvm-openmp.sh

```diff
@@ -10,10 +10,16 @@ if [[ "${target_platform}" == "linux"* ]]; then
   export LDFLAGS="$LDFLAGS -static-libgcc -static-libstdc++"
 fi
 
+if [ "$(uname)" == "Darwin" ]; then
+  EXTRA_ARGS='$EXTRA_ARGS -DCMAKE_C_FLAGS="-mlinker-version=305"'
+  EXTRA_ARGS='$EXTRA_ARGS -DCMAKE_CXX_FLAGS="-mlinker-version=305"'
```

Comment:
Do we need a variable for this or to set it as a function of the llvm version or the version of some other package? 

I have a vague memory of having to parse this info out of the output of some other tool a while ago...

Reply:
Hmm, as I understood it is more an OSX version issue with llvm 10.0.0.  The configuration parameter a choosen depending on host's OSX version ...

See for some details here https://stackoverflow.com/questions/60934005/clang-10-fails-to-link-c-application-with-cmake-on-macos-10-12

Reply:
Sure, but that linker behalfs OSX version dependent ...

Reply:
Ugh. So this version matches our ci? If the version is OS X dep we should parse it from ld-v so we don’t have to change it later.

Reply:
As it seems to be issued by clang's driver ... as far as I understood is clang version 9.x still working as expected ... we might want to add those flags to clang-compiler-activation-feedstock.

Reply:
The culprit thing is here in clang's driver at https://github.com/llvm/llvm-project/blob/llvmorg-10.0.0/clang/lib/Driver/Driver.cpp#L400-L409
You see that it uses the linker version used on built time. Result of ld -v is not considered here.

Reply:
Sure that seems ok. cc @conda-forge/core what do you all think?

Reply:
Please use `$CFLAGS` and `$CXXFLAGS` to append these as using `-DCMAKE_C_FLAGS` directly would disable the `CFLAGS` from the environment.

Reply:
I think if we set the correct `CMAKE_LINKER` in the clangdev recipe, then this issue should go away.

Reply:
I modified it so it appends option ot CFLAGS, and CXXFLAGS instead.

Reply:
As far as I understood seems CMAKE_LINKER not doing what we want here. We still want to use modern cctools, but just wanna to control the linker version to be used.  But it might I missed here something about this option

Reply:
What's the exact error that you are getting?

Reply:
After building llvm/clang 10, with recent cctools-and-ld64 package already present, I ran into this issue during building openmp by using clang 10 bootstrap. Remark here is that llvm/clang uses here of course cf's 9.x bootstrap version.
So new linker is sure present, but clang used